### PR TITLE
Sprint 01 - register function

### PR DESCRIPTION
**Pull request by Trinhnguyen1704**
Create register API in back-end and register function in front-end.

### CODE REVIEW

File: DEMO-PROJECT-FE/src/axios-base.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import axios from 'axios';
+
+const instance = axios.create({
+  baseURL: 'https://localhost:5001/'
```
Comment (vancuong196) [inline DEMO-PROJECT-FE/src/axios-base.js:4]: can we put this base url in env configuration?

File: DEMO-PROJECT-FE/src/actions/registerAction.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    dispatch(registerRequest());
+    const registerData = {
+      email: email,
+      password: password,
+      fullName: fullName,
+      phoneNumber: phoneNumber
+    };
+    try{
+        console.log(registerData);
+        const response = await axios.post("/api/register/", registerData);
```
Comment (vancuong196) [inline DEMO-PROJECT-FE/src/actions/registerAction.js:36]: create constant for API endpoint.

File: DEMO-PROJECT-BE/DP.Repository/Implement/Auth/RegisterRepository.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                _context.AccountHasRoles.Add(accountHasRole);
+                _context.SaveChanges();
+                
+                return account.Id;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(JsonConvert.SerializeObject(e));
+            }
+            return -1;
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP.Repository/Implement/Auth/RegisterRepository.cs:65]: Make a class like CreateAccountResult { int accountId, bool success}

File: DEMO-PROJECT-FE/src/screens/Register/RegisterScreen.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    const [emailError, setEmailError] = useState("");
+    const [phoneNumber, setPhoneNumber] = useState("");
+    const [passwordError, setPasswordError] = useState("");
+    const [fullNameError, setFullNameError] = useState("");
+    const [phoneNumberError, setPhoneNumberError] = useState("");
+
+    const onTextBoxPassword = (event) => {
+        if(event.target.value ==="")
+        {
+            setPasswordError("Password is required!");
```
Comment (vancuong196) [inline DEMO-PROJECT-FE/src/screens/Register/RegisterScreen.js:22]: Put all string into a file as constants

File: DEMO-PROJECT-BE/DP.Repository/Implement/Auth/RegisterRepository.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                    Code = "USERCODE"
+                };
+
+                _context.Accounts.Add(registerAccount);
+                _context.SaveChanges();
+
+                var account = _context.Accounts.FirstOrDefault(x => x.Email == register.Email);
+                
+                //Add profile
+                var profile = _context.Profiles.FirstOrDefault(x => x.AccountId == account.Id);
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP.Repository/Implement/Auth/RegisterRepository.cs:46]: Where is profile created?

Comment (vancuong196) [reply DEMO-PROJECT-BE/DP.Repository/Implement/Auth/RegisterRepository.cs:46]: is the profile null?

**Review by vancuong196 — CHANGES_REQUESTED** [review]

File: DEMO-PROJECT-BE/DP.Repository/Implement/Auth/RegisterRepository.cs (hunk 5 above)
Comment (Trinhnguyen1704) [reply DEMO-PROJECT-BE/DP.Repository/Implement/Auth/RegisterRepository.cs:46]: The profile is not null. When I create account, a profile with this account id is created too, with no information.

**Pull request merged**